### PR TITLE
Clickhouse: type mapping change and cleanup

### PR DESCRIPTION
**Pull request by Amogh-Bharadwaj**
- Fix syncing of Timestamp and TimestampTZ
- Map Timestamps and Dates to `Nullable(type)` because epoch in most cases not a sane default.
- Remove dead code

### CODE REVIEW

**Review by iskakaushik — APPROVED** [review]

File: flow/connectors/clickhouse/normalize.go [hunk 1]
```diff
@@ -81,21 +81,23 @@ func generateCreateTableSQLForNormalizedTable(
...
+		case qvalue.QValueKindNumeric:
 			precision, scale := numeric.ParseNumericTypmod(column.TypeModifier)
 			if column.TypeModifier == -1 || precision > 76 || scale > precision {
 				precision = numeric.PeerDBClickhousePrecision
 				scale = numeric.PeerDBClickhouseScale
 			}
 			stmtBuilder.WriteString(fmt.Sprintf("`%s` DECIMAL(%d, %d), ",
 				colName, precision, scale))
-		} else {
+		case qvalue.QValueKindTimestamp, qvalue.QValueKindTimestampTZ, qvalue.QValueKindDate:
```
Comment (serprex) [inline flow/connectors/clickhouse/normalize.go:93]: Can we track nullable in QValue?

Comment (Amogh-Bharadwaj) [reply flow/connectors/clickhouse/normalize.go:93]: That, and the issue specifically with clickhouse is that [Nullable affects performance and storage](https://clickhouse.com/docs/en/cloud/bestpractices/avoid-nullable-columns) and so even once we wire up nullable in qvalue, clients may not necessarily want columns to be Nullable

File: flow/connectors/clickhouse/normalize.go [hunk 2]
```diff
@@ -176,17 +178,24 @@ func (c *ClickhouseConnector) NormalizeRecords(ctx context.Context, req *model.N
 			if err != nil {
 				return nil, fmt.Errorf("error while converting column type to clickhouse type: %w", err)
 			}
-			if clickhouseType == "DateTime64(6)" || clickhouseType == "UUID" {
+			if clickhouseType == "UUID" {
```
Comment (serprex) [inline flow/connectors/clickhouse/normalize.go:181]: Wondering if UUID can be extracted with "String"

Comment (serprex) [reply flow/connectors/clickhouse/normalize.go:181]: Tested: it can

File: flow/connectors/clickhouse/normalize.go [hunk 3]
```diff
@@ -176,17 +178,24 @@ func (c *ClickhouseConnector) NormalizeRecords(ctx context.Context, req *model.N
...
 				clickhouseType = "String"
 			}
```
Comment (serprex) [inline flow/connectors/clickhouse/normalize.go:183]:
```suggestion
```

File: flow/connectors/clickhouse/normalize.go (hunk 1 above)
Comment (serprex) [reply flow/connectors/clickhouse/normalize.go:93]: yes, but that should be an option, ideally per-column

in this case it'd only be used for timestamps to be nullable when they're nullable

**Comment by Amogh-Bharadwaj** [comment]
functionally tested

**Review by serprex — APPROVED** [review]

**Pull request merged**